lib/perl/Genome/Model/Tools/DetectVariants2/Filter/FalsePositiveVcfBase.pm

```diff
@@ -621,81 +619,50 @@ sub generate_and_run_readcounts_in_parallel {
-            left_property=>"reference_fasta",
-            right_operation=>$op,
-            right_property=>"reference_fasta",
-        );
-        $workflow->add_link(
-            left_operation=>$workflow->get_input_connector,
-            left_property=>"region_list",
-            right_operation=>$op,
-            right_property=>"region_list",
-        );
-        $workflow->add_link(
-            left_operation=>$workflow->get_input_connector,
-            left_property=>"use_version",
-            right_operation=>$op,
-            right_property=>"use_version",
-        );
-        $workflow->add_link(
-            left_operation=>$workflow->get_input_connector,
-            left_property=>"minimum_base_quality",
-            right_operation=>$op,
-            right_property=>"minimum_base_quality",
+            command => "Genome::Model::Tools::Sam::Readcount",
         );
-        $workflow->add_link(
-            left_operation=>$workflow->get_input_connector,
-            left_property=>"bam_$sample",
-            right_operation=>$op,
-            right_property=>"bam_file",
+        $workflow->add_operation($op);
+        for my $property (qw(reference_fasta region_list use_version minimum_base_quality)) {
+            $workflow->connect_input(
+                input_property => $property,
+                destination => $op,
+                destination_property => $property,
+            );
+        }
+
+        $workflow->connect_input(
+            input_property => "bam_$sample",
+            destination => $op,
+            destination_property => "bam_file",
         );
-        $workflow->add_link(
-            left_operation=>$workflow->get_input_connector,
-            left_property=>"readcounts_$sample",
-            right_operation=>$op,
-            right_property=>"output_file",
+        $workflow->connect_input(
+            input_property => "readcounts_$sample",
+            destination => $op,
+            destination_property => "output_file",
         );
-        $workflow->add_link(
-            left_operation=>$op,
-            left_property=>"output_file",
-            right_operation=>$workflow->get_output_connector,
-            right_property=>"output",
+        $workflow->connect_output(
+            source => $op,
+            source_property => "output_file",
+            output_property => "output_$sample",
         );
     }
 
     my $log_dir = $self->output_directory;
-    if(Workflow::Model->parent_workflow_log_dir) {
-        $log_dir = Workflow::Model->parent_workflow_log_dir;
-    }
-    $workflow->log_dir($log_dir);
-
-    my @errors = $workflow->validate;
-    if (@errors) {
-        $self->error_message(@errors);
-        die "Errors validating workflow\n";
+    if(my $parent_dir = Genome::WorkflowBuilder::DAG->parent_log_dir) {
+        $log_dir = $parent_dir;
     }
+    $workflow->recursively_set_log_dir($log_dir);
 
-    Genome::Sys->disconnect_default_handles;
```

Comment:
Yes; it's one of the things that is now inside `Genome::WorkflowBuilder::DAG` for both [Workflows](https://github.com/genome/genome/blob/e6fc9e92729a38ab9b68b07ce584ecfbc889273a/lib/perl/Genome/WorkflowBuilder/DAG.pm#L166) and [PTero executions](https://github.com/genome/genome/blob/e6fc9e92729a38ab9b68b07ce584ecfbc889273a/lib/perl/Genome/WorkflowBuilder/DAG.pm#L191).